historicalData/historicalDataRepository.ts

```diff
@@ -0,0 +1,156 @@
+import type { Logger } from '@aws-lambda-powertools/logger'
+import {
+	QueryCommand,
+	type TimestreamQueryClient,
+} from '@aws-sdk/client-timestream-query'
+import {
+	Context,
+	HistoricalDataRequest,
+	HistoricalDataResponse,
+	chartProto,
+} from '@hello.nrfcloud.com/proto/hello'
+import { parseResult } from '@nordicsemiconductor/timestream-helpers'
+import { Type, type Static } from '@sinclair/typebox'
+import { Value } from '@sinclair/typebox/value'
+import { groupBy } from 'lodash-es'
+import { getQueryStatement } from './queryGenerator.js'
+
+export type HistoricalRequest = Omit<
+	Static<typeof HistoricalDataRequest>,
+	'data'
+>
+type HistoricalResponse = Static<typeof HistoricalDataResponse>
+
+export const transformTimestreamData = (
+	data: Record<string, unknown>[],
+	mapKeys: { fromKey: string; toKey: string }[],
+): Record<string, unknown>[] => {
+	const transformedData = []
+
+	// Added ts to mapKeys as default
+	mapKeys.push({ fromKey: 'time', toKey: 'ts' })
+
+	const groupedData = groupBy(data, (d) => d.time)
+	for (const item in groupedData) {
+		const data = groupedData[item]
+		const tranformedRecord = data?.reduce<Record<string, unknown>>(
```

Comment:
Please don't mix programming styles, you use both `reduce()` and `for ... in`. Pick one.

Reply:
`groupBy` returns `{ [index: string]: T[] }`. So, there is no native `forEach` function to iterate Dictionary